src/utils/getTokenList.ts

```diff
@@ -89,6 +90,40 @@ export default async function getTokenList(
   throw new Error('Unrecognized list URL protocol.')
 }
 
+// loop to fetch all whitelist token
+export async function getTokenListV2(
+  listUrl: string,
+  resolveENSContentHash?: (ensName: string) => Promise<string>,
+): Promise<TokenList> {
+  return new Promise(async (resolve, reject) => {
+    let tokens: any[] = []
+    try {
+      const pageSize = 100
+      const maximumPage = 15
+      let page = 1
+      while (true) {
+        const { data } = await axios.get(`${listUrl}&pageSize=${pageSize}&page=${page}`)
```

Comment:
Why set chunk size = 100, and maximumPage = 15?

Reply:
this is fetching whitelist token, currently it is  < 100 tokens per chain => so this loop maybe loop 1 or 2 time.    15 page is enough for us, it only use prevent infinity loop when api return wrong data